### PR TITLE
update chain-deps and implement storage error type inside jormungandr

### PR DESCRIPTION
**Pull request by eugene-babichenko**

### CODE REVIEW

File: jormungandr/src/blockchain/chain.rs [hunk 1]
```diff
@@ -50,6 +50,7 @@ See Internal documentation for more details: doc/internal_design.md
 */
 
 use super::{branch::Branches, reference_cache::RefCache};
+use crate::blockchain::StorageError;
 use crate::{
     blockcfg::{
         Block, Block0Error, BlockDate, ChainLength, Epoch, EpochRewardsInfo, Header, HeaderHash,
```
Comment (NicolasDP) [inline jormungandr/src/blockchain/chain.rs:56]: if you are to more the import, put it in the same _scope_ of all the `use crate::{*}` just below

File: jormungandr/src/rest/v0/logic.rs [hunk 2]
```diff
@@ -6,6 +6,7 @@
 // - All errors should be processed on the framework  integration side. Usually
 //   they are 400 or 500.
 
+use crate::blockchain::StorageError;
 use crate::{
     diagnostic::Diagnostic,
```
Comment (NicolasDP) [inline jormungandr/src/rest/v0/logic.rs:11]:
```suggestion
use crate::{
    blockchain::StorageError,
    diagnostic::Diagnostic,
```

File: jormungandr/src/blockchain/chain.rs (hunk 1 above)
Comment (eugene-babichenko) [reply jormungandr/src/blockchain/chain.rs]: done

File: jormungandr/src/rest/v0/logic.rs (hunk 2 above)
Comment (eugene-babichenko) [reply jormungandr/src/rest/v0/logic.rs]: done

**Pull request merged**